src/components/textInput/TextInputBase.tsx

```diff
@@ -0,0 +1,348 @@
+  secretInput?: boolean;
+  onValidate?: (value: string) => boolean;
+  errorMessage?: string;
+}
+
+const styles = StyleSheet.create({
+  helperRow: {
+    ...IOStyles.row,
+    alignItems: "center",
+    paddingHorizontal: 10
+  },
+  textInput: {
+    ...IOStyles.row,
+    alignItems: "center",
+    paddingVertical: 8,
+    height: 60,
+    borderRadius: 8,
+    paddingHorizontal: 12
+  },
+  textInputStyle: {
+    fontSize: 16,
+    marginTop: IOSpacingScale[2],
+    fontWeight: "400",
+    lineHeight: 24,
+    fontFamily: "Readex Pro",
+    height: "100%"
+  },
+  textInputLabelWrapper: {
+    position: "absolute",
+    paddingHorizontal: 10,
+    zIndex: 10,
+    bottom: 0,
+    top: 0,
+    justifyContent: "center"
+  }
+});
+
+type InputTextHelperRow = {
+  value: string;
+  counterLimit?: number;
+  helperText?: string;
+  errorMessage?: string;
+  isValid?: boolean;
+};
+
+const HelperRow = ({
+  value,
+  counterLimit,
+  helperText,
+  errorMessage,
+  isValid
+}: InputTextHelperRow) => {
+  const valueCount = useMemo(() => value.length, [value]);
+
+  const helperRowStyle: ViewStyle = useMemo(() => {
+    if (counterLimit && helperText) {
+      return {
+        justifyContent: "space-between"
+      };
+    }
+    if (counterLimit) {
+      return {
+        justifyContent: "flex-end"
+      };
+    }
+    if (helperText) {
+      return {
+        justifyContent: "flex-start"
+      };
+    }
+    return {};
+  }, [counterLimit, helperText]);
+
+  return (
+    <View
+      style={[
+        IOStyles.row,
+        {
+          alignItems: "center",
+          paddingHorizontal: 10
+        },
+        helperRowStyle
+      ]}
+    >
+      {errorMessage && isValid === false && (
+        <LabelSmall weight="Regular" color="error-600">
+          {errorMessage}
+        </LabelSmall>
+      )}
+      {helperText && !(errorMessage && isValid === false) && (
+        <LabelSmall weight="Regular" color="grey-700">
+          {helperText}
+        </LabelSmall>
+      )}
+      {counterLimit && (
+        <LabelSmall
+          weight="Regular"
+          color="grey-700"
+        >{`${valueCount} / ${counterLimit}`}</LabelSmall>
+      )}
+    </View>
+  );
+};
+
+export const TextInputBase = ({
+  disabled = false,
+  placeholder,
+  value = "",
+  icon,
+  counterLimit,
+  helperText,
+  onChangeText,
+  onValidate,
+  secretInput,
+  errorMessage
+}: InputTextProps) => {
+  const labelSharedValue = useSharedValue(0);
+  const [inputStatus, setInputStatus] = React.useState<InputStatus>(
+    disabled ? "disabled" : "initial"
+  );
+  const [isValid, setIsValid] = React.useState<boolean | undefined>(undefined);
+  const [isSecretInput, setSecretInput] = React.useState<boolean>(
+    secretInput ?? false
+  );
+
+  const inputRef = useRef<TextInput>(null);
+
+  const boxStyle: ViewStyle = useMemo(() => {
+    if (inputStatus === "focused") {
+      return {
+        borderColor: IOColors["blueIO-500"],
+        borderWidth: 2
```

Comment:
Changing the `borderWidth` on the focus state results in a slight visual glitch on the contained text (it moves a bit):

https://github.com/pagopa/io-app-design-system/assets/1255491/aeec7c40-1630-42d0-ac62-34fcd6714eb5

It's a minor issue, but I'd try to use `boxShadow` property instead (with zero blur, used as substitute of the border)

Reply:
Unfortunately it's not possible to remove the blur from the shadow in RN, I just tested it on iOS but as far as I remember shadow actually acts in a different way between android and iOS

This is the max result i got, but it's far way different from the mock up in Figma.

<img src="https://github.com/pagopa/io-app-design-system/assets/3959405/fd243af5-2ce8-4303-8729-3058f60cb0ae" height="700"/>